descriptor/qsarmolecular/src/main/java/org/openscience/cdk/qsar/descriptors/molecular/CarbonTypesDescriptor.java

```diff
@@ -154,12 +155,12 @@ public DescriptorValue calculate(IAtomContainer container) {
         int c4sp3 = 0;
 
         for (IAtom atom : container.atoms()) {
-            if (!atom.getSymbol().equals("C") && !atom.getSymbol().equals("c")) continue;
```

Comment:
@johnmay - Flagging this line, because I wasn't sure of the additional condition of matching a lowercase c in `&& !atom.getSymbol().equals("c")`. Can elements be referenced with casing characters?

Reply:
Yes this can be replaced, the lower case c is from SMILES for aromatic carbon. In SMARTS it would be ``[!C!c]`` but you can simplify that to just ``[!#6]``